platform/x11/detect.py

```diff
@@ -145,6 +146,14 @@ def configure(env):
     env.ParseConfig('pkg-config xcursor --cflags --libs')
     env.ParseConfig('pkg-config xrandr --cflags --libs')
 
+    if (env['touch'] == 'yes'):
+        x11_error = os.system("pkg-config xi --modversion > /dev/null ")
+        if (x11_error):
+            print("xi not found.. cannot build with touch. Aborting.")
+            sys.exit(255)
+        env.ParseConfig('pkg-config xi --cflags --libs')
+        env.Append(CPPFLAGS=['-DTOUCH_ENABLED'])
```

Comment:
Do we need to make it optional? It's true that it adds a dependency on Xi, but since we depend on X11 already I think it's not an issue. Mostly asking because our compiler command line is huge already so I'm very wary of adding new CPPFLAGS :P

Reply:
I added that as a safety measure. The code performs a grab on all direct touch devices to avoid interference from OS gesture detection. That should be OK, but since it goes beyond what other projects are doing I though allowing it to be disabled was safer by now.

Reply:
Ok, fine by me.